monai/transforms/compose.py

```diff
@@ -41,7 +41,116 @@
 
 logger = get_logger(__name__)
 
-__all__ = ["Compose", "OneOf", "RandomOrder", "SomeOf", "execute_compose"]
+__all__ = [
+    "Compose",
+    "OneOf",
+    "RandomOrder",
+    "SomeOf",
+    "transform_iterator",
+    "ranged_transform_iterator",
+    "execute_compose",
+]
+
+
+def transform_iterator(compose, step_into_all=False):
+    """
+    ``compose_iterator`` is a function that returns an iterator over the transforms in a ``Compose``
+    instance.
+
+    It works by iterating over transforms, and stepping into nested ``Compose`` instances in a
+    recursive fashion. Given the following example:
+
+    ..code-block:: python
+
+    ::
+        t1, t2, t3, t4 = # some transforms
+        c = Compose([t1, Compose([t2, t3]), t4])
+        print([tx for tx in compose_iterator(c)])
+
+        > [t1, t2, t3, t4]
+
+    By default, ``compose_iterator`` doesn't step into ``OneOf``, ``SomeOf``, ``RandomOrder`` or
+    other classes that inherit ``Compose``. If ``step_into_all`` is passed True, it will step into
+    any Compose-like object as if it were just a container. As such, if you are using ``compose_iterator``
+    to step over objects as if executing them, ``step_into_all`` should be False. If you are using
+    ``compose_iterator`` to step over the static structure of the pipeline (such as if you are setting
+    or clearing flags on every transform), you should set ``step_into_all`` to True
+
+    Args:
+        compose: A `Compose` instance that contains the transforms to be iterated over
+        step_into_all: A boolean flag that indicates whether to step into randomised `Compose`
+            sub-classes such as `OneOf`, `SomeOf`, and `RandomOrder`
+    """
+
+    if isinstance(compose, (list, tuple)):
+        transforms = compose
+    elif type(compose) is Compose or (step_into_all is True and isinstance(compose, Compose)):
```

Comment:
this is not a pythonic way of handling the compose and its other potential subclasses. let's not merge this into core.

Reply:
It's pretty much how flatten operates, so its no less pythonic than what is already in the repo

Reply:
https://github.com/Project-MONAI/MONAI/blob/fbbb64b4cb2facffccbf4fdf6c44e20f8d31ca10/monai/transforms/compose.py#L322